### PR TITLE
Fix pthread cmdline issue

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
- fs/procfs: Avoid the duplicated 0x prefix in pthread cmdline
- pthread: Change the default name from <pthread> to <0xyyyyyyyy>
Before change:
```
   63    20 100 RR       pthread --- Waiting  Signal    00000000 006148 000744  12.1%    0.0% <pthread> 0x0x341de68c
```
After change:
```
   63    20 100 RR       pthread --- Waiting  Signal    00000000 006148 000744  12.1%    0.0% <0x34187c00> 0x341de68c
```
Note, you can still change pthread name through pthread_setname_np, this patch just modify the default behaviour.

## Impact
The content from: /proc/xxx/cmdline

### CODE REVIEW

File: include/pthread.h [hunk 1]
```diff
@@ -195,11 +195,8 @@
  * error number in errno.  This is an inconsistency with the pthread interfaces.
  */
 
-#define pthread_setname_np(thread, name) \
-  prctl((int)PR_SET_NAME_EXT, (char*)name, (int)thread)
-
-#define pthread_getname_np(thread, name) \
-  prctl((int)PR_GET_NAME_EXT, (char*)name, (int)thread)
+#define pthread_setname_np(thread, name) prctl(PR_SET_NAME_EXT, name, thread)
+#define pthread_getname_np(thread, name) prctl(PR_GET_NAME_EXT, name, thread)
```
Comment (yamt) [inline include/pthread.h:199]: is this change necessary? i feel it's safer to be explicit about types for vararg functions.

Comment (xiaoxiang781216) [reply include/pthread.h:199]: Ok, I will drop the last commit from PR

**Comment by davids5** [comment]
Will you consider a "pt-" prefix?

**Comment by xiaoxiang781216** [comment]
Is it enough to identify pthread by <>? task/kthread doesn't quote his name by <> normally.

**Comment by davids5** [comment]
My feeling is that before it said what it was now it does not.  The prefix of `pt` at least is a clue. Not really more obvious than <> but `pthread-0xyyyyy` would tell the whole story.

**Comment by xiaoxiang781216** [comment]
Ok, it show pt-0xyyyyyyyy now.

**Review by davids5 — APPROVED** [review]
